pydantic_ai_slim/pydantic_ai/durable_exec/temporal/_model.py

```diff
@@ -190,3 +235,74 @@ async def request_stream(
     def _validate_model_request_parameters(self, model_request_parameters: ModelRequestParameters) -> None:
         if model_request_parameters.allow_image_output:
             raise UserError('Image output is not supported with Temporal because of the 2MB payload size limit.')
+
+    def _get_model_id(self, model: models.Model | models.KnownModelName | str | None = None) -> str | None:
+        """Get the model ID for the given model parameter.
+
+        Returns a string that will be checked against registered model IDs,
+        or passed to infer_model if not found. Returns None to use the default model.
+        """
+        if model is None or model == 'default':
+            return None
+
+        if isinstance(model, Model):
+            # Check if this model instance is already registered
+            for model_id, registered_model in self._models_by_id.items():
+                if registered_model is model:
+                    if model_id == 'default':
+                        return None
+                    return model_id
+            raise UserError(
+                'Arbitrary model instances cannot be used at runtime inside a Temporal workflow. '
+                'Register the model via `models` or reference a registered model by ID.'
+            )
+
+        return model
+
+    def resolve_model_for_outside_workflow(
+        self, model: models.Model | models.KnownModelName | str | None = None
+    ) -> models.Model | models.KnownModelName | str | None:
+        """Resolve a model parameter for use outside a workflow.
+
+        If the model is a registered ID, returns the registered Model instance.
+        If model is None, returns the wrapped (primary) model to use as default.
+        Otherwise returns the original value.
+        """
+        if model is None:
+            # Return the primary model (wrapped model or first registered)
+            return self.wrapped
```

Comment:
I think we can drop this if/return, because returning `None` is fine from the perspective of the caller